src/sortStudents.ts

```diff
@@ -1,16 +1,49 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getMiddleValue(people: number[]): number {
+  return people.reduce((sum, item) => sum + item, 0) / people.length;
+}
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? +a[sortBy] - +b[sortBy]
+          : +b[sortBy] - +a[sortBy];
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? getMiddleValue(a[sortBy]) - getMiddleValue(b[sortBy])
+          : getMiddleValue(b[sortBy]) - getMiddleValue(a[sortBy]);
+      default:
+        throw new Error('Invalid type data');
```

Comment:
It makes your code more readable
```suggestion
  return [...students].sort((a, b) => {
    switch (sortBy) {
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? a[sortBy].localeCompare(b[sortBy])
          : b[sortBy].localeCompare(a[sortBy]);
          
      case SortType.Age:
      case SortType.Married:
        return order === 'asc'
          ? +a[sortBy] - +b[sortBy]
          : +b[sortBy] - +a[sortBy];
          
      case SortType.AverageGrade:
        return order === 'asc'
          ? getMiddleValue(a[sortBy]) - getMiddleValue(b[sortBy])
          : getMiddleValue(b[sortBy]) - getMiddleValue(a[sortBy]);
          
      default:
        throw new Error('Invalid type data');
```